projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid.component.ts

```diff
@@ -424,25 +434,40 @@ export class IgxTreeGridComponent extends IgxGridBaseComponent {
                     }
                 }
             }
-       } else {
+        } else {
             const record = this.records.get(rowID);
-            const childData = record.parent ? record.parent.data[this.childDataKey] : this.data;
+            const childData = record.parent ? record.parent.data[this.childDataKey] :
+                this.transactions.enabled ? this._gridAPI.get_all_data(this.id, true) : this.data;
```

Comment:
This can at least be split into two variables so it's readable, if it can't be optimized with `this._gridAPI.get_all_data(this.id, this.transactions.enabled)` since it's not the original record ref.

Reply:
yes, we can go here without added rows. This will put delete transaction for root add row with undefined recordRef but we do not need it in this case.